addons/web/static/src/legacy/js/public/public_root_instance.js

```diff
@@ -15,19 +15,52 @@ import {
 } from "../../utils";
 import { ComponentAdapter } from "web.OwlCompatibility";
 
-
 import { registry } from "@web/core/registry";
-import { makeEnv, startServices } from "../../../env";
+import { loadBundleTemplates } from "@web/core/assets";
+import { makeEnv, startServices } from "@web/env";
+import { ErrorHandler, NotUpdatable } from "@web/core/utils/components";
 const serviceRegistry = registry.category("services");
 
-const { Component } = owl;
+const { Component, tags, mount } = owl;
 
 /**
  * Configure Owl with the public env
  */
 owl.config.mode = legacyEnv.isDebug() ? "dev" : "prod";
 owl.Component.env = legacyEnv;
 
+class MainComponentsContainer extends Component {
```

Comment:
i wonder if it would make sense to create that component in webclient, so we do not duplicate code here

Reply:
Oh yeah, I intended to do that, I was just trying this out to see if it would work and forgot to factor it out of the webclient template and just use this component instead.